app/controllers/email_subscriptions_controller.rb

```diff
@@ -82,12 +44,30 @@ def update
   end
 
   def destroy
-    email_subscription.destroy! unless is_transition_checker_subscription?
+    email_subscription.destroy!
     head :no_content
   end
 
 private
 
+  # remove this after we've done a bulk import of older subscriptions
+  def migrate_transition_checker_subscription
+    return unless params.fetch(:subscription_name) == TRANSITION_CHECKER_SUBSCRIPTION_NAME
+    return if @email_subscription
```

Comment:
This is wrong, should be `return if email_subscription` - I think a test for the case where the subscription has already been migrated would catch this.